src/platform_impl/macos/app_state.rs

```diff
@@ -654,7 +651,7 @@ impl AppState {
         let wait_timeout = HANDLER.wait_timeout(); // configured by pump_events
         let app_timeout = match HANDLER.control_flow() {
             ControlFlow::Wait => None,
-            ControlFlow::Poll | ControlFlow::ExitWithCode(_) => Some(Instant::now()),
```

Comment:
So I think this is handled a couple of lines above?
```rust
if HANDLER.is_exit() {
    Self::stop();
}
```
But have no clue actually.